app/src/ui/diff/modified-image-diff.tsx

```diff
@@ -1,25 +1,342 @@
+            {this.state.naturalHeightBefore}px
+          </div>
+        </div>
+        <div className="image-diff__after">
+          <div className="image-diff__header">Added</div>
+          {renderImage(this.props.current, {
+            onLoad: this.handleImgLoadAfter,
+            style,
+          })}
+          <div className="image-diff__footer">
+            <span className="strong">W:</span> {this.state.naturalWidthAfter}px
+            | <span className="strong">H:</span> {this.state.naturalHeightAfter}px
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  private renderDifference(
+    height: number,
+    width: number,
+    widthContainer: number
+  ) {
+    return (
+      <div
+        className="image-diff_inner--difference"
+        style={{
+          height,
+          width,
+          left: (widthContainer - PADDING - width) / 2 + PADDING / 2,
+        }}
+      >
+        <div className="image-diff__before">
+          {renderImage(this.props.previous, {
+            onLoad: this.handleImgLoadBefore,
+            style: {
+              maxHeight: height,
+              maxWidth: width,
+            },
+          })}
+        </div>
+        <div className="image-diff__after">
+          {renderImage(this.props.current, {
+            onLoad: this.handleImgLoadAfter,
+            style: {
+              maxHeight: height,
+              maxWidth: width,
+              mixBlendMode: 'difference',
+            },
+          })}
+        </div>
+      </div>
+    )
+  }
+
+  private renderFade(height: number, width: number, widthContainer: number) {
+    const style = {
+      height,
+      width,
+    }
+    return (
+      <div
+        className="image-diff_inner--fade"
+        style={{
+          ...style,
+          marginBottom: 30,
+          left: (widthContainer - PADDING - width) / 2 + PADDING / 2,
+        }}
+      >
+        <div className="image-diff__before" style={style}>
+          {renderImage(this.props.previous, {
+            onLoad: this.handleImgLoadBefore,
+            style: {
+              maxHeight: height,
+              maxWidth: width,
+            },
+          })}
+        </div>
+        <div
+          className="image-diff__after"
+          style={{
+            ...style,
+            opacity: this.state.value,
+          }}
+        >
+          {renderImage(this.props.current, {
+            onLoad: this.handleImgLoadAfter,
+            style: {
+              maxHeight: height,
+              maxWidth: width,
+            },
+          })}
+        </div>
+        <input
+          style={{ margin: `${height + 10}px 0 0 ${(width - 129) / 2}px` }}
+          type="range"
+          max={1}
+          min={0}
+          value={this.state.value}
+          step={0.001}
+          onChange={this.handleValueChange}
+        />
+      </div>
+    )
+  }
+
+  private renderSwipe(height: number, width: number, widthContainer: number) {
```

Comment:
I wonder if these different render methods should be split out into their own components?